clang/test/CAS/cached-diagnostics.c

```diff
@@ -0,0 +1,64 @@
+// RUN: rm -rf %t
+// RUN: split-file %s %t/src
+// RUN: mkdir %t/out
+
+// RUN: %clang_cc1 -triple x86_64-apple-macos12 -fsyntax-only %t/src/main.c -I %t/src/inc -Wunknown-pragmas 2> %t/regular-diags1.txt
+
+// RUN: %clang -cc1depscan -fdepscan=inline -fdepscan-include-tree -fdepscan-prefix-map=%t/src=/^src -o %t/t1.rsp -cc1-args \
+// RUN:   -cc1 -triple x86_64-apple-macos12 -fcas-path %t/cas -faction-cache-path %t/cache \
+// RUN:     -emit-obj %t/src/main.c -o %t/out/output.o -I %t/src/inc -Wunknown-pragmas
+
+// Compare diagnostics after a miss.
+// RUN: %clang @%t/t1.rsp 2> %t/diags1.txt
+// RUN: diff -u %t/regular-diags1.txt %t/diags1.txt
+
+// Check that we have both the remark and source diagnostics.
+// RUN: %clang @%t/t1.rsp -Rcompile-job-cache 2> %t/diags-hit1.txt
+// RUN: FileCheck %s -input-file %t/diags-hit1.txt
+// CHECK: remark: compile job cache hit for
+// CHECK: warning: some warning
+// CHECK: warning: unknown pragma ignored
+// CHECK: warning: using the result of an assignment as a condition without parentheses
```

Comment:
Should we also test driver diagnostics? Since the merging is alway a corner case.

Reply:
Merging driver diagnostics into serialized diagnostics is tested via https://github.com/apple/llvm-project/blob/next/clang/test/CAS/fcache-compile-job-serialized-diagnostics.c#L39